client/src/pages/Cart.tsx

```diff
@@ -1,14 +1,154 @@
-import React from "react";
+import React, { ChangeEvent, Fragment, useEffect, useState } from "react";
 import Layout from "../components/Layout";
 
+import PageTitle from "../components/PageTitle";
+import CartListItem from "../components/CartListItem";
+import TotalPriceWithButton from "../components/TotalPriceWithButton";
+import { RootState } from "../redux/reducers";
+import { useSelector } from "react-redux";
+import { useActions } from "../hooks/useActions";
+import { PATH } from "../constants";
+import { useNavigate } from "react-router-dom";
+import { OrderDetail } from "../types/dto";
+
 const Cart = () => {
+  const [checkAll, setCheckAll] = useState(true);
+  const [checkboxes, setCheckboxes] = useState<boolean[]>();
+  const [quantities, setQuantities] = useState<number[]>();
+  const [totalPrice, setTotalPrice] = useState(0);
+
+  const navigation = useNavigate();
+
+  const { getCartProducts, deleteCartProduct, postPaymentProducts } =
+    useActions();
+  const { products } = useSelector((state: RootState) => state.carts);
+
+  useEffect(() => {
+    getCartProducts();
+  }, [getCartProducts]);
+
+  useEffect(() => {
+    if (products) {
+      setCheckboxes(products.map(() => true));
+      setQuantities(products.map(() => 1));
+    }
+  }, [products]);
+
+  useEffect(() => {
+    if (products && checkboxes && quantities) {
+      const totalPrice = products.reduce(
+        (acc, cur, idx) =>
+          (acc = checkboxes[idx]
+            ? acc + cur.product.price * quantities[idx]
+            : acc),
```

Comment:
순간ㅋㅋㅋㅋ `acc = checkboxes[idx]` 가 어떻게 조건문이 되지!>!?!?! 라고 뇌정지가 ... 죄송합니다
callbackFn 의 반환값이 acc 니까 아래처럼 해도 되지 않을까영?
```suggestion
          checkboxes[idx]
            ? acc + cur.product.price * quantities[idx]
            : acc,
```